### DISCUSSION TITLE
Updates: How to get updates

### DISCUSSION THREAD

**Discussion by Kliton** [gotd/td#573] [Q&A] [answered]
There are apis for getting updates of client?

I haven't found any example

**Comment by ghost** [comment, marked as answer]
You need to set an `UpdateHandler` field in client options ([`telegram.Options`](https://pkg.go.dev/github.com/gotd/td/telegram#Options)).

Also, there is an [`tg.UpdateDispatcher`](https://github.com/gotd/td/blob/main/tg/tl_handlers_gen.go#L34) type for dispatching updates by type.

Echo bot example: https://github.com/gotd/td/blob/main/examples/bot-echo/main.go#L37-L47

Also look at gap resolver example: https://github.com/gotd/td/blob/main/examples/updates-gap-recover/main.go

Similiar question: https://github.com/gotd/td/issues/463#issuecomment-897529315

**Comment by Kliton** [comment]
Thanks @tdakkota 

The problem that i had is that i'm saving the update pts etc on my database ( postgres ) because i'm checking updates every 5 minutes.

So i need a method that gives me all the updates since a certain point.

There is this kind of method in the lib?

Thank you

**Comment by Kliton** [comment]
I think i found out:

```	err = client.Run(context.Background(), func(ctx context.Context) error {
		state, err := client.API().UpdatesGetState(ctx)
		if err != nil {
			return err
		}

		fmt.Println("Getting update with pts", state.Pts)

		updates, err := client.API().UpdatesGetDifference(ctx, &tg.UpdatesGetDifferenceRequest{
			Pts:           state.Pts,
			PtsTotalLimit: 200,
			Date:          state.Date,
			Qts:           state.Qts,
		})
		if err != nil {
			return err
		}

		switch v := updates.(type) {
		case *tg.UpdatesDifferenceEmpty: // updates.differenceEmpty#5d75a138
			fmt.Println("There are no updates")
			return nil
		case *tg.UpdatesDifference: // updates.difference#f49ca0
			fmt.Println("Updates difference", len(v.OtherUpdates))
			return nil
		case *tg.UpdatesDifferenceSlice: // updates.differenceSlice#a8fb1981
			fmt.Println("Update difference slices", len(v.OtherUpdates))
			return nil
		case *tg.UpdatesDifferenceTooLong: // updates.differenceTooLong#4afe8f6d
			fmt.Println("Updates too long", v.Pts)
			return nil
		}

		fmt.Println("Got upddates differnce")

		return nil
	})```

**Reply by ernado** [reply]
This is not supported in gap resolver, but is possible by calling api directly.
Unfortunately it will be pretty complex :(
